fedimint-client/src/lib.rs

```diff
@@ -522,6 +596,23 @@ impl IGlobalClientContext for ModuleGlobalClientContext {
     async fn transaction_update_stream(&self) -> BoxStream<OperationState<TxSubmissionStates>> {
         self.client.transaction_update_stream(self.operation).await
     }
+
+    async fn log_event_raw(
+        &self,
+        dbtx: &mut ClientSMDatabaseTransaction<'_, '_>,
+        kind: EventKind,
+        module: Option<(ModuleKind, ModuleInstanceId)>,
+        payload: serde_json::Value,
+    ) {
```

Comment:
Well.. it's a different kind of raw here. I can't add generic version of this function (`P : Serialize`) because it would make this trait not object safe. So I need the caller to pass json. I will rename to `log_event_json`.

Reply:
Well, you could just keep it bytes-only till the end. Using `Value` as an itermediate state is probably also quite allocation-heavy. Or as you mentioned: always dealing with JSON and never with bytes. If that's the only nit left in the end I'll happily merge though xD

Reply:
Letting modules emit random bytes events, while assuming everything is json seems like inviting problems. Yeah, the perf of it is meh, but event log shouldn't be used for heavy throughput events and these payloads should be small, so I think it's OK. Eventually this should just go away.